### PR TITLE
Bonus entity, repository and service have been created.

### PR DESCRIPTION
**Pull request by andrycreator**

### CODE REVIEW

File: src/main/java/com/familytime/model/service/implementation/BonusImpl.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+import com.familytime.model.entity.Bonus;
+import com.familytime.model.repository.BonusRepository;
+import com.familytime.model.service.BonusService;
+
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.List;
+
+public class BonusImpl implements BonusService {
```
Comment (vtsutsman) [inline src/main/java/com/familytime/model/service/implementation/BonusImpl.java]: @andrycreator rename `BonusImpl` to `BonusServiceImpl`.

File: src/main/java/com/familytime/model/service/implementation/BonusImpl.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    BonusRepository bonusRepository;
+
+    /**
+     * Find bonus by ID.
+     * @param id unique identificatos
+     * @return Bonus found
+     */
+    @Override
+    public Bonus findById(Long id) {
+        return null;
```
Comment (vtsutsman) [inline src/main/java/com/familytime/model/service/implementation/BonusImpl.java]: @andrycreator Why do U return `null`?

File: src/main/java/com/familytime/model/service/implementation/BonusImpl.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    /**
+     * Delete existed bonus.
+     * @param id unique identificator bonus
+     */
+    @Override
+    public void delete(Long id) {
+
```
Comment (vtsutsman) [inline src/main/java/com/familytime/model/service/implementation/BonusImpl.java]: @andrycreator Add implementation.

File: src/main/java/com/familytime/model/service/implementation/BonusImpl.java (hunk 1 above)
Comment (vtsutsman) [inline src/main/java/com/familytime/model/service/implementation/BonusImpl.java]: @andrycreator Here should be annotation `@Service`

File: src/main/java/com/familytime/model/repository/BonusRepository.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import com.familytime.model.entity.Bonus;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.transaction.annotation.Transactional;
+
+/**
+ * Interface for work with persistence layout
+ * @version 1.0
+ */
+@Transactional
```
Comment (vtsutsman) [inline src/main/java/com/familytime/model/repository/BonusRepository.java]: @andrycreator U should not use `@Transactional` on the `repository` level.

**Review by vtsutsman — CHANGES_REQUESTED** [review]
Add implementation of service's methods.
Use `@Transactional` only in `service` layer.

File: src/main/java/com/familytime/model/service/implementation/BonusImpl.java (hunk 1 above)
Comment (andrycreator) [reply src/main/java/com/familytime/model/service/implementation/BonusImpl.java]: The changes have been done.

Comment (andrycreator) [reply src/main/java/com/familytime/model/service/implementation/BonusImpl.java]: The changes have been done.

File: src/main/java/com/familytime/model/service/implementation/BonusImpl.java (hunk 2 above)
Comment (andrycreator) [reply src/main/java/com/familytime/model/service/implementation/BonusImpl.java]: The changes have been done.

File: src/main/java/com/familytime/model/service/implementation/BonusImpl.java (hunk 3 above)
Comment (andrycreator) [reply src/main/java/com/familytime/model/service/implementation/BonusImpl.java]: The changes have been done.

File: src/main/java/com/familytime/model/repository/BonusRepository.java (hunk 4 above)
Comment (andrycreator) [reply src/main/java/com/familytime/model/repository/BonusRepository.java]: The changes have been done.

**Comment by andrycreator** [comment]
All changes have been done, please take a look.

**Review by vtsutsman — APPROVED** [review]
Looks good to me.

**Review by vtsutsman — APPROVED** [review]

**Review by Valentine1996 — APPROVED** [review]

**Pull request merged**